### DISCUSSION TITLE
SQL Server connection Quartz Server

### DISCUSSION THREAD

**Discussion by apoellitsi1926** [quartznet/quartznet#1573] [Q&A]
Hi everyone,

I am facing an issue with the Quartz server installation as a service. I want my Quartz to use SQL server. 
I downloaded Quartz.NET-3.4.0 and copied at my ProgramFiles folder the libraries etc from bin/release/QuartzServer and at the quartz.config file i added the following parameters:

`# this section configures the scheduler to use SQL Server as the jobstore
quartz.jobStore.type = Quartz.Impl.AdoJobStore.JobStoreTX, Quartz
quartz.jobStore.driverDelegateType = Quartz.Impl.AdoJobStore.SqlServerDelegate, Quartz
quartz.jobStore.tablePrefix = QRTZ_
quartz.jobStore.clustered = false
quartz.jobStore.useProperties = false

quartz.jobStore.misfireThreshold=60000
quartz.jobStore.dataSource = default
quartz.dataSource.default.provider = SqlServer-20
quartz.dataSource.default.connectionString = Server=localhost\SQLEXPRESS;Database=*******;UID=sa;PWD=*****`

When i try to run the service i get the error that the service can not start.

If i remove the above code from the config file the service starts immediately.

Please bear in mind that i am using Quartz since 2016 and i was using the same params and everything was working.

**Comment by lahma** [comment]
You should probably try to start the same service as console app to see what is the error / enable logging to file. Service runners like [Topshelf](https://github.com/Topshelf/Topshelf) allow you to run both as console and Windows service for example.

**Reply by apoellitsi1926** [reply]
Hi based on the console app i am getting the following error:

`ERROR Quartz.Server.QuartzServer - Server initialization failed:Could not Initialize DataSource: default
Quartz.SchedulerException: Could not Initialize DataSource: default
 ---> System.ArgumentOutOfRangeException: There is no metadata information for provider 'SqlServer-20' (Parameter 'providerName')
   at Quartz.Impl.AdoJobStore.Common.DbProvider.GetDbMetadata(String providerName) in /_/src/Quartz/Impl/AdoJobStore/Common/DbProvider.cs:line 122`

**Reply by apoellitsi1926** [reply]
Do i need to add something? Am i missing something?

**Reply by lahma** [reply]
Try using just "SqlServer" instead of "SqlServer-20", like [described in migration guide](https://www.quartz-scheduler.net/documentation/quartz-3.x/migration-guide.html#simplified-job-store-provider-names).

**Reply by apoellitsi1926** [reply]
Thank you for your reply. I tried that one also but i am getting a different error:

`System.ArgumentException: Error while reading metadata information for provider 'SqlServer' (Parameter 'providerName')
 ---> Quartz.SchedulerConfigException: Could not parse property 'connectionType' into correct data type: Could not load file or assembly 'Microsoft.Data.SqlClient, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
 ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Data.SqlClient, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.`

**Reply by lahma** [reply]
If you are using the old System client library, then it's `SystemDataSqlClient` , [here's the mapping](https://github.com/quartznet/quartznet/blob/main/src/Quartz/Impl/AdoJobStore/Common/dbproviders.netstandard.properties). Other option is to add NuGet reference to [Microsoft.Data.SqlClient](https://www.nuget.org/packages/Microsoft.Data.SqlClient/).

**Reply by apoellitsi1926** [reply]
Hi,

As i said I downloaded Quartz.NET-3.4.0 and copied at my ProgramFiles(x86) folder the libraries/dll etc from bin/release/QuartzServer/net462 and at the quartz.config file i added the parameters in order to use Sql Server.

I haven't build or change anything to the project. i just did a copy paste like i did in 2015.
